receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -347,9 +369,135 @@ func (r *pReceiver) initPrometheusComponents(ctx context.Context, logger log.Log
 			r.settings.TelemetrySettings.ReportStatus(component.NewFatalErrorEvent(err))
 		}
 	}()
+
 	return nil
 }
 
+func (r *pReceiver) initPrometheusAPI(ctx context.Context, host component.Host) error {
+	r.settings.Logger.Info("Starting Prometheus API server")
+
+	o := &web.Options{
+		ScrapeManager: r.scrapeManager,
+		Context:       ctx,
+		ListenAddress: r.cfg.PrometheusAPIServer.ServerConfig.Endpoint,
+		ExternalURL: &url.URL{
+			Scheme: "http",
+			Host:   r.cfg.PrometheusAPIServer.ServerConfig.Endpoint,
+			Path:   "",
+		},
+		RoutePrefix:    "/",
+		ReadTimeout:    time.Minute * readTimeoutMinutes,
+		PageTitle:      "Prometheus Receiver",
+		Flags:          make(map[string]string),
+		MaxConnections: maxConnections,
+		IsAgent:        true,
+		Gatherer:       prometheus.DefaultGatherer,
```

Comment:
have you tested this with multiple receivers?  We've had collisions between metrics in different receivers in the past when using the global gatherer.  You might need to use r.registerer, or make a gatherer specific to this receiver.

Reply:
Thanks @dashpole I added a test for this with multiple receivers with it enabled. It uses the r.registerer below this but it looks like the DefaultGatherer doesn't have an equivalent function to just add the label  like the registerer and is for having a whole different implementation. I made sure it works with both using prometheus.DefaultGatherer but different registerers